apis/v1alpha2/grpcroute_types.go

```diff
@@ -0,0 +1,552 @@
+	//
+	// Support in GRPCRouteRule: Core
+	//
+	// Support in GRPCBackendRef: Extended
+	GRPCRouteFilterRequestHeaderModifier GRPCRouteFilterType = "RequestHeaderModifier"
+
+	// GRPCRouteFilterRequestHeaderModifier can be used to add or remove a gRPC
+	// header from a gRPC response before it is sent to the client.
+	//
+	// Support in GRPCRouteRule: Core
+	//
+	// Support in GRPCBackendRef: Extended
+	GRPCRouteFilterResponseHeaderModifier GRPCRouteFilterType = "ResponseHeaderModifier"
+
+	// GRPCRouteFilterRequestMirror can be used to mirror gRPC requests to a
+	// different backend. The responses from this backend MUST be ignored by
+	// the Gateway.
+	//
+	// Support in GRPCRouteRule: Extended
+	//
+	// Support in GRPCBackendRef: Extended
+	GRPCRouteFilterRequestMirror GRPCRouteFilterType = "RequestMirror"
+
+	// GRPCRouteFilterExtensionRef should be used for configuring custom
+	// gRPC filters.
+	//
+	// Support in GRPCRouteRule: Implementation-specific
+	//
+	// Support in GRPCBackendRef: Implementation-specific
+	GRPCRouteFilterExtensionRef GRPCRouteFilterType = "ExtensionRef"
+)
+
+// GRPCRouteFilter defines processing steps that must be completed during the
+// request or response lifecycle. GRPCRouteFilters are meant as an extension
+// point to express processing that may be done in Gateway implementations. Some
+// examples include request or response modification, implementing
+// authentication strategies, rate-limiting, and traffic shaping. API
+// guarantee/conformance is defined based on the type of the filter.
+type GRPCRouteFilter struct {
+	// Type identifies the type of filter to apply. As with other API fields,
+	// types are classified into three conformance levels:
+	//
+	// - Core: Filter types and their corresponding configuration defined by
+	//   "Support: Core" in this package, e.g. "RequestHeaderModifier". All
+	//   implementations supporting GRPCRoute MUST support core filters.
+	//
+	// - Extended: Filter types and their corresponding configuration defined by
+	//   "Support: Extended" in this package, e.g. "RequestMirror". Implementers
+	//   are encouraged to support extended filters.
+	//
+	// - Implementation-specific: Filters that are defined and supported by specific vendors.
+	//   In the future, filters showing convergence in behavior across multiple
+	//   implementations will be considered for inclusion in extended or core
+	//   conformance levels. Filter-specific configuration for such filters
+	//   is specified using the ExtensionRef field. `Type` MUST be set to
+	//   "ExtensionRef" for custom filters.
+	//
+	// Implementers are encouraged to define custom implementation types to
+	// extend the core API with implementation-specific behavior.
+	//
+	// If a reference to a custom filter type cannot be resolved, the filter
+	// MUST NOT be skipped. Instead, requests that would have been processed by
+	// that filter MUST receive a HTTP error response.
```

Comment:
Which error?

Reply:
@gnossen any ideas for what we should do here?